weather_app_functions.R

```diff
@@ -0,0 +1,234 @@
+### weather_app in R Shiny ###
+
+library(httr)
+library(jsonlite)
+library(dplyr)
+library(leaflet)
+library(mapview)
+library(magick)
+library(tidygeocoder)
+
+# tidygeocoder::geo_osm() uses a string with an address as input and returns
+# the latitude and longitude of the location. 
+
+# you will need an API key to access the data from openweathermaps.org 
+# you have to sign up for a free account to get the api key 
+# the api key is private and should not be shared with other people
+
+my_api_key <- "insert your api key here" 
+
+
+# The following function takes the location (in string format) and your apiKey (in string format) as input.
+# The function returns the weather data for a specific location as a list.
+# The weather data are provided by www.openweathermap.org
+
+get_weather <- function(location, apiKey) {
+  
+  # first, we need to use tidygeocoder::geo_osm to get the longitude and the latitude of the desired location. 
+  
+  latitude <- tidygeocoder::geo_osm(location)$lat[1]
+  longitude <- tidygeocoder::geo_osm(location)$long[1]

```

Comment:
In order to prevent error messages and give nice feedback to the user, check for latitude/longitude == NA in case location cannot be geocoded.